internal/outpost/proxyv2/filesystemstore/filesystemstore.go

```
@@ -0,0 +1,185 @@
+	if err != nil {
+		if syscall.EWOULDBLOCK.Is(err) {
+			return ErrSessionCleanupAlreadyRunning
+		}
+		return err
+	}
+	defer func() {
+		if flockErr := syscall.Flock(int(lockFile.Fd()), syscall.LOCK_UN); flockErr != nil {
+			s.log.WithError(flockErr).Warn("failed to unlock file")
+		}
+	}()
+
+	done := make(chan error, 1)
+	go func() {
+		done <- s.sessionCleanup(ctx)
+	}()
+
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case err = <-done:
+		s.log.WithField("error", err).Info("Session cleanup finished")
+		return err
+	}
+}
+
+// sessionCleanup checks the modification time of all session files and removes them
+// when they reach the configured maximum age in the session store.
+// Since the FilesystemStore from Gorilla does not have a session cleanup function,
+// it is only necessary for the filesystem session store.
+func (s *Store) sessionCleanup(ctx context.Context) error {
+	files, err := os.ReadDir(s.storePath)
+	if err != nil {
+		return err
+	}
+
+	var lastErr error
+	for _, file := range files {
+		select {
+		case <-ctx.Done():
+			s.log.Warn("session cleanup interrupted during file processing")
+			return ctx.Err()
+		default:
+		}
+
+		if !strings.HasPrefix(file.Name(), "session_") {
+			continue
+		}
+
+		fullPath := path.Join(s.storePath, file.Name())
+		stat, err := os.Lstat(fullPath)
+		if err != nil {
+			s.log.WithError(err).WithField("path", fullPath).Warning("failed to read stats from file")
+			lastErr = err
+			continue
+		}
+
+		modTime := stat.ModTime()
+		if time.Since(modTime) <= time.Duration(s.Options.MaxAge)*time.Second {
+			s.log.WithField("max-age", s.Options.MaxAge).WithField("modified", modTime.String()).Debug("session still valid")
+			continue
+		}
+
+		s.log.WithField("path", fullPath).WithField("modified", modTime.String()).Info("cleanup expired session")
+		if err = os.Remove(fullPath); err != nil {
+			s.log.WithError(err).WithField("path", fullPath).Warn("failed to delete session")
+			lastErr = err
```

Using lastErr, the last error encountered is returned. All others are silently dropped. If multiple deletions fail, you'd never know. I would figure it's best to collect all the errors with errors.Join probably doing something like this:
```
var errs []error
...
errs = append(errs, err)
...
return errors.Join(errs...)
```

---

Good idea, I hadn't thought about that and therefore included the errors in the log. I have now improved it.